docs/2-building-blocks-of-oop-part-1/hometask/src/Triangle.ts

```diff
@@ -1,3 +1,36 @@
-import { Shape } from './Shape';
+import { Shape } from './Shape'
+import { Point } from './Point';
 
-export class Triangle extends Shape {}
+export class Triangle extends Shape {
+  constructor(pointA: Point, pointB: Point, pointC: Point);
+  constructor(pointA: Point, pointB: Point, pointC: Point, color: string, filled: boolean);
+  constructor(pointA: Point, pointB: Point, pointC: Point, color?: string, filled?: boolean) {
+    if (typeof color === 'string' && typeof filled === 'boolean') {
+      super([pointA, pointB, pointC], color, filled);
+    } else {
+      super([pointA, pointB, pointC])
+    }
+  }
+
+  toString() {
+    let pointsStrArr: string[] = []
+    for (let i = 0; i < this.points.length; i++) {
+      pointsStrArr.push(`v${i + 1}=${this.points[i].toString()}`);
+    }
+    return `Triangle[${pointsStrArr.join(',')}]`;
+  }
+  
+  getType() {
+    const sideLengthArray: number[] = [];
+    for (let i = 0; i < this.points.length; i++) {
+      const prevPoint = i
+        ? this.points[i - 1]
+        : this.points[this.points.length - 1];
+      const sideLength = this.points[i].distance(prevPoint);
+      if (!sideLengthArray.includes(sideLength)) {
+        sideLengthArray.push(sideLength);
+      }
+    }
+    return sideLengthArray.length === 2 ? 'isosceles triangle' : sideLengthArray.length === 1 ? 'equilateral triangle' : 'scalene triangle';
+  }
```

Comment:
Hi, the logic overall is fine, but when we should consider more about the design, the `getType` sounds like a simple action and may return value quickly, it should not contain complexity in it like `distance` calculation, I would suggest create a data member like `private type: TriangleType`, the value of `type` is defined as the Object is created(the logic can be moved to here). Every time you want to `getType` is just to `return this.type` simply. Thanks.